libvirt/resource_libvirt_domain_test.go

```diff
@@ -176,6 +176,49 @@ func TestAccLibvirtDomain_VolumeTwoDisks(t *testing.T) {
 	})
 }
 
+func TestAccLibvirtDomain_VolumeCache(t *testing.T) {
+	var domain libvirt.Domain
+	var volumeQCOW2 libvirt.StorageVol
+	randomVolumeQCOW2 := acctest.RandString(10)
+	randomDomainName := acctest.RandString(10)
+
+	var config = fmt.Sprintf(`
+	resource "libvirt_volume" "%s" {
+		name = "%s"
+		format = "qcow2"
+	}
+
+	resource "libvirt_domain" "%s" {
+		name = "%s"
+		disk {
+			volume_id = "${libvirt_volume.%s.id}"
+			cache = "unsafe"
+		}
+	}`, randomVolumeQCOW2, randomVolumeQCOW2, randomDomainName, randomDomainName, randomVolumeQCOW2)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckLibvirtDomainDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckLibvirtDomainExists("libvirt_domain."+randomDomainName, &domain),
+					testAccCheckLibvirtVolumeExists("libvirt_volume."+randomVolumeQCOW2, &volumeQCOW2),
+					// Check that each disk has the appropriate cache
```

Comment:
with this comment we state `each` disk but we are just testing one disk.

We should test that `cache` is nil for the normal or remove the comment because is contradictory